cro-clib/src/fee.rs

```diff
@@ -0,0 +1,58 @@
+use crate::types::get_string;
+use crate::types::CroResult;
+use crate::types::{CroFee, CroFeePtr};
+pub use chain_core::init::network::Network;
+use chain_core::tx::fee::{LinearFee, Milli};
+use std::ffi::CString;
+use std::os::raw::c_char;
+use std::ptr;
+use std::str::FromStr;
+
+/// create fee
+#[no_mangle]
+/// # Safety
+pub unsafe extern "C" fn cro_create_fee(
+    fee_out: *mut CroFeePtr,
+    constant_string: *const c_char,
+    coeff_string: *const c_char,
+) -> CroResult {
+    let base = get_string(constant_string);
+    let coeff = get_string(coeff_string);
+    let fee = CroFee {
+        fee: LinearFee::new(
+            Milli::from_str(&base).expect("read milli from constant"),
+            Milli::from_str(&coeff).expect("read milli from coefficient"),
+        ),
+    };
+    let fee_box = Box::new(fee);
+    ptr::write(fee_out, Box::into_raw(fee_box));
+    CroResult::success()
+}
+
+/// estimate fee
+#[no_mangle]
+/// # Safety
+pub unsafe extern "C" fn cro_estimate_fee(
+    fee_ptr: CroFeePtr,
+    length: u32,
+    fee_output: *mut u8,
+    fee_output_length: u32,
```

Comment:
Fee is an opaque type around Coin/u64 and has a fixed length.

For C bindings, the equivalent should ideally use C-style opaque types (something like `struct fee { uint64_t x; }`). 

(a more messy approach is to take `*mut u64` directly, but one will lose compile-time checking for APIs that, for example, may take amount + fee where one may by mistake switch the arguments)

Reply:
it will just return u64, i'll modify